Modified team list component to view/create teams

sanketbansal opened the pull request:
<!--
(Thanks for sending a pull request! .)
-->

<!--The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor." -->

<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->


#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- Modified the UI to look similar to the current EvalAI application.
- Changes made in logic to create and select the host team to create challenge.
- Changes made in loader UI to show status of the application processing.
- Changes made in logic to create/ select team for participation in challenge.

<!-- Demo Link: Add here the link where you changes can be seen. -->
- Link to live demo: http://pr-149-evalai.surge.sh <!-- Replace XXX with your PR no: -->

<!-- Screenshots for the change: Add here the screenshot of the fix. -->
![Screenshot from 2019-06-16 10-44-43](https://user-images.githubusercontent.com/17106489/59565856-3f7fae00-9026-11e9-9d99-4031927ade97.png)

![Screenshot from 2019-06-16 10-44-56](https://user-images.githubusercontent.com/17106489/59565862-49091600-9026-11e9-8a38-045e04564ed1.png)

sanketbansal committed:
Ported image assets

Ported the image assets from angularJs application into 'image-port' directory

sanketbansal committed:
Ported image assets

Ported the image assets from angularJs application into 'image-port' directory

sanketbansal committed:
Added styles from angularjs

Ported styles from angularjs Application inside style-port folder

sanketbansal committed:
Added styles from angularjs

Ported styles from angularjs Application inside style-port folder

sanketbansal committed:
Modified footer component

Added github fork and star button using github api

sanketbansal committed:
Modified footer component

Added github fork and star button using github api

sanketbansal committed:
Modified auth service

Changes made in auth services to integrate angularJS authentication features

sanketbansal committed:
Modified auth service

Changes made in auth services to integrate angularJS authentication features

sanketbansal committed:
Changes in app component

Modified app component to show loader according to authService 'isLoader' variable.

sanketbansal committed:
Changes in app component

Modified app component to show loader according to authService 'isLoader' variable.

sanketbansal committed:
Modified app loader component

app loader component modified to accompany AngularJS application UI.

sanketbansal committed:
Modified app loader component

app loader component modified to accompany AngularJS application UI.

sanketbansal committed:
Modified angular.json file

Added main.scss file of ported style in the angular.json file to compile it at the time of building project.

sanketbansal committed:
Modified angular.json file

Added main.scss file of ported style in the angular.json file to compile it at the time of building project.

sanketbansal committed:
Removed routing from auth service

Routing to 'auth/login' has been removed

sanketbansal committed:
Removed routing from auth service

Routing to 'auth/login' has been removed

sanketbansal committed:
Added side bar component

Added side bar as utility component for the side bar in the Dashboard UI

sanketbansal committed:
Added dash footer component

Dash footer added for improving dashboard UI

sanketbansal committed:
Added dash footer component

Dash footer added for improving dashboard UI

sanketbansal committed:
changes in style files

Added font variables in ported variable file
Added font style in ported main file.

sanketbansal committed:
changes in style files

Added font variables in ported variable file
Added font style in ported main file.

sanketbansal committed:
Modified footer

Made the footer UI similar to the angularjs application one.

sanketbansal committed:
Modified footer

Made the footer UI similar to the angularjs application one.

sanketbansal committed:
Added simple header component

simple header added to modify the UI of the header in the dashboard

sanketbansal committed:
Modifications in simple header component

Added color property for signup button
Added styles for login button

sanketbansal committed:
Modified card list component

Removed team card logic to decouple the teamlist component logic.

sanketbansal committed:
Modified app component

Removed ngIf  condition from app-loading component to resolve the error:
ExpressionChangedAfterItHasBeenCheckedError

sanketbansal committed:
Modified app component

Removed ngIf  condition from app-loading component to resolve the error:
ExpressionChangedAfterItHasBeenCheckedError

sanketbansal committed:
Modified teamlist component

Modfied the UI to resemble  with current EvalAI apllication
Added logic to select or create participant team
Added logic to select or create host team.
Added team card component directly by removing dependency over card-list component.
Added logic to participate in challenege by selecting existing participant team.

sanketbansal committed:
Modified team card component

Modify the logic to integrate it directly into the teamlist component.
Modify the UI by using styles of both current evalAI and previous evalAIngx project.
Fixed bug in team card's test bed configuration

sanketbansal committed:
Changes in team list component

Added authentication logic inside team list  component to navigate to login page if the user is not logged in.

sanketbansal committed:
Fixed testing issue

Added FormsModule in tesBed configuration to use ngForms

sanketbansal commented:
@lunayach @Shekharrajak please review pr

Shekharrajak commented:
I didn't understand why there are 162 file changes. Is there any irrelevant commits (not part of this PR) as well ?

sanketbansal committed:
Added feature to start stop loader

Added methods which are used by the components to start and stop the loader.

sanketbansal committed:
Moved styles in teamlist component

Styles related to team-list component moved from 'styles-port/modules/teams.scss.

sanketbansal committed:
Moved styles in teamlist component

Styles related to team-list component moved from 'styles-port/modules/teams.scss.

sanketbansal committed:
Fixed bugs after rebasing

Bugs fixed after rebasing the PR with the master

sanketbansal committed:
Fixed bugs after rebasing

Bugs fixed after rebasing the PR with the master

sanketbansal committed:
Changes in team card component

Modifed selected style in team card component
Fixed bug in the UI

sanketbansal committed:
Removed duplicate styles

Removed duplicate scss files from basic.scss

sanketbansal committed:
Removed duplicate styles

Removed duplicate scss files from basic.scss

sanketbansal committed:
Removed assets and styles

Unused assets and styles removed from images-port folder and styles-port folder

sanketbansal committed:
Removed assets and styles

Unused assets and styles removed from images-port folder and styles-port folder

sanketbansal committed:
Added styles

Aded base and main.scss files

sanketbansal committed:
Added styles

Aded base and main.scss files

sanketbansal committed:
Added basic styles

Basic style files added in styles-port folder

sanketbansal committed:
Added basic styles

Basic style files added in styles-port folder

sanketbansal committed:
Changes in project configuration

Added basic.scss file globally in angular.json 
Removed duplicate imports from app module file

sanketbansal committed:
Changes in project configuration

Added basic.scss file globally in angular.json 
Removed duplicate imports from app module file

sanketbansal committed:
Removed main.scss

main.scss file removed from styles-port folder

sanketbansal committed:
Removed main.scss

main.scss file removed from styles-port folder

sanketbansal committed:
Added component styles

Added component specific styles

sanketbansal committed:
Added component styles

Added component specific styles

sanketbansal committed:
Removed logs

Removed console logs and commets

sanketbansal committed:
Removed logs

Removed console logs and commets

sanketbansal committed:
changes in base.scss

Modified styles in base.scss file

Shekharrajak commented:
Before checking the new commits. First I look into my previous comment and I don't see reply for it.

sanketbansal commented:
> I didn't understand why there are 162 file changes. Is there any irrelevant commits (not part of this PR) as well ?
I have created the PR to reduce file changes as there is dependency among PRs.

sanketbansal committed:
Fixed bug in the UI

Added styles to fix the bug in simple header component.

sanketbansal committed:
Changes after rebasing

Removed unused styles and component folder
Modified angular.json file
Fixed bug in teamlist component

sanketbansal committed:
Changes after rebasing

Removed unused styles and component folder
Modified angular.json file
Fixed bug in teamlist component

lunayach commented on src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts:
```diff
@@ -155,11 +154,11 @@ export class TeamcardComponent implements OnInit, OnChanges {
       this.isSelected = false;
     }
-    if (this.selected) {
+    /*if (this.selected) {
```
Remove this.

sanketbansal replied:
Done

lunayach commented on src/styles/base.scss:
```diff
@@ -1484,8 +1485,20 @@ td, th {
+  }
+}
+
+.btn-pagination:hover {
```
This is not creating an effect. As styles with and without **: hover** are same.

lunayach commented on src/app/components/publiclists/teamlist/teamlist.component.ts:
```diff
@@ -248,7 +281,8 @@ export class TeamlistComponent implements OnInit {
     }
-    self.allTems = temp;
+    self.allTeams = temp;
+    // console.log('LOGI teamlist unselectOtherTeam', self.allTeams);
```
Remove this.

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/teamlist/teamlist.component.ts:
```diff
@@ -269,41 +303,21 @@ export class TeamlistComponent implements OnInit {
     this.apiService.getUrl(path).subscribe(
       data => {
+        this.globalService.stopLoader();
+        // console.log(data['results']);
```
Also this.

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/teamlist/teamlist.component.ts:
```diff
@@ -449,31 +463,34 @@ export class TeamlistComponent implements OnInit {
+          this.globalService.stopLoader();
+          this.globalService.handleFormError(this.components, err);
+        },
+        () => console.log('CREATE-TEAM-FINISHED')
```
Please check for these statements in the whole PR.

sanketbansal replied:
done

lunayach reviewed (changes requested):
Looks good, just a few minor changes are required.

lunayach commented:
Two more things:

- I am not seeing the T&C's modal before registering from a team like in the current version.
- The label on the **Add** button seems misaligned.
![ui](https://user-images.githubusercontent.com/21227893/61814712-367ec980-ae7b-11e9-8048-edd49c2f9b14.png)

sanketbansal committed:
Changes in teamlist component

Added modal for terms and conditions
Removed unnecessary changes

sanketbansal commented:
> Two more things:
> 
> * I am not seeing the T&C's modal before registering from a team like in the current version.
> * The label on the **Add** button seems misaligned.
>   ![ui](https://user-images.githubusercontent.com/21227893/61814712-367ec980-ae7b-11e9-8048-edd49c2f9b14.png)

Done

sanketbansal committed:
Fixed UI bug

Added css in base.scss file to align text of the button.

lunayach commented on src/app/components/publiclists/teamlist/teamlist.component.ts:
```diff
@@ -269,46 +333,23 @@ export class TeamlistComponent implements OnInit {
    */
   fetchTeams(path) {
     const SELF = this;
+    this.globalService.startLoader('Fetching Teams');
```
@sanketbansal when do you see this? I think I was not able to view it.

sanketbansal replied:
Basically, it gets viewed when the loader starts on fetching data.
In your case it is fetching data locally very quick and that's why you might not be able to see it.

lunayach committed:
Merge branch 'master' into modified-teamList

lunayach commented on src/app/components/publiclists/teamlist/teamlist.component.html:
```diff
@@ -1,91 +1,63 @@
+                          (deleteTeamCard)="deleteTeamWrapper()($event)" (editTeamCard)="editTeamWrapper()($event)" (addMembersTeamCard)="addMembersToTeamWrapper()($event)"
+            (selectTeamCard)="selectTeamWrapper()($event)">
+            </app-teamcard>
+          </ul>
```
When I go to say, 2nd page in pagination i.e. **(Page 2 of 2)**, all teams from the previous pages are also displayed with the remaining teams of this page.

sanketbansal replied:
okay, correcting it

sanketbansal replied:
done

lunayach committed:
Merge branch 'master' into modified-teamList

sanketbansal committed:
Changes in teamlist component

Show particular teams on pagination

lunayach commented on src/app/components/publiclists/teamlist/teamlist.component.html:
```diff
@@ -1,91 +1,63 @@
-                    </div>
-                  </div>
-                </div>
+<section class="ev-sm-container ev-view ">
```
Minor: When I create a team having the name same as one of the already existing team, it does not create the team, nor does it throws any error toast. Maybe an appropriate error would make sense here.

sanketbansal replied:
Done

lunayach commented on src/app/components/publiclists/teamlist/teamlist.component.ts:
```diff
@@ -449,58 +490,65 @@ export class TeamlistComponent implements OnInit {
-      SELF.challengeService.participateInChallenge(SELF.challenge['id'], SELF.selectedTeam['id']);
-    };
+    const confirmCallback = () => { this.challengeService.participateInChallenge(this.challenge['id'], this.selectedTeam['id']); };
+
```
`\n`'s are redundant here. `ol` and `li` are block elements by default.

sanketbansal replied:
Done

lunayach commented on src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts:
```diff
@@ -26,7 +26,7 @@ export class TeamcardComponent implements OnInit, OnChanges {
    * Update
    */
-  @Input() update: any;
+  @Input() isOnChallengePage: any;
```
```suggestion
  @Input() isOnChallengePage: boolean;
```

sanketbansal replied:
Done

lunayach commented on src/app/components/challenge/challengeparticipate/challengeparticipate.component.spec.ts:
```diff
@@ -11,6 +11,7 @@ import { RouterTestingModule } from '@angular/router/testing';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
 import { TeamlistComponent } from '../../../components/publiclists/teamlist/teamlist.component';
 import { EndpointsService } from '../../../services/endpoints.service';
+import {FormsModule} from '@angular/forms';
```
```suggestion
import { FormsModule } from '@angular/forms';
```
(Inconsistent spacing. To avoid it in future, change your editor's `.ts` settings)

sanketbansal replied:
Done

lunayach reviewed (changes requested):
Functionality is working. Minor changes required.

lunayach committed:
Merge branch 'master' into modified-teamList

sanketbansal committed:
Changes in teamlist component

Added toast mesage for unsuccesful team creatiion.

lunayach reviewed (approved):
@RishabhJain2018 Looks good to be merged!
